CHANGELOG.md

```diff
@@ -1,5 +1,42 @@
 # Changelog
 
+## v0.31.4
+
+*April 12th, 2019*
+
+This release fixes a regression from v0.31.3 which used the peer's `SocketAddr` to add the peer to
+the address book. This swallowed the peer's self-reported port which is important in case of reconnect.
+It brings back `NetAddress()` to `NodeInfo` and uses it instead of `SocketAddr` for adding peers.
+Additionally, it improves response time on the `/validators` or `/status` RPC endpoints.
+As a side-effect it makes these RPC endpoint more difficult to DoS and fixes a performance degradation in `ExecCommitBlock`.
+
+Special thanks to external contributors on this release:
+@brapse, @guagualvcha, @mydring
+
+### BREAKING CHANGES:
+
+* Go API
+  - [p2p] [\#3545](https://github.com/tendermint/tendermint/pull/3545) The `AddrBook` interface method `MarkAsGood` now only takes a `p2p.ID` instead of a `p2p.NetAddress`
+  - [p2p] [\#3011](https://github.com/tendermint/tendermint/pull/3011) Remove `ListOfKnownAddresses` from the `AddrBook` interface
```

Comment:
p2p isn't listed in https://github.com/tendermint/tendermint/#semantic-versioning so technically we don't need to document these :)

Reply:
removed in https://github.com/tendermint/tendermint/pull/3550